jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -665,6 +667,7 @@ public void run()
             try
             {
                 createEndPoint(channel, key);
+                _selectorManager.onAccepted(channel,EndPoint.class.cast(key.attachment()));
```

Comment:
I think these 2 lines should be swapped, otherwise `open` listeners are called _before_ `accepted` listeners.

Reply:
These lines are deliberately in this order.   We want the call to onAccepted to happen after the endpoint is created and opened, so the onAccepted handler can see the endPoint and Connection that have been allocated to the accepted connection.   Also if we called onAccepted before open, then the total count would drop by 1 before going back up, which is not the behaviour that we want.

Reply:
Why `onAccepted` should see the `EndPoint` at all ? I would remove the parameter.
The second part of your reasoning is telling me that we are adding a listener based on only one of the possible implementations we may have, so the listener is not generic and will only work for one case.
This listener is about the "accept" phase of a connection as it is processed by our NIO implementation. It is very counter-intuitive that a connection is accepted after the open event. In HTTP/2 we write quite some data from onOpen(), and yet the connection won't be accepted yet. If the connection is closed from onOpen(), then we have a close event _before_ the accepted event. I don't think it's what it should be.